metrics_collector/volume_collector.go

```diff
@@ -80,26 +160,63 @@ func (vc *VolumeCollector) Describe(ch chan<- *prometheus.Desc) {
 func (vc *VolumeCollector) Collect(ch chan<- prometheus.Metric) {
 	defer func() {
 		if err := recover(); err != nil {
-			vc.logger.WithField("error", err).Warn("panic during collecting metrics")
+			vc.logger.WithField("error", err).Warn("Panic during collecting metrics")
 		}
 	}()
 
 	volumeLists, err := vc.ds.ListVolumesRO()
 	if err != nil {
-		vc.logger.WithError(err).Warn("error during scrape ")
+		vc.logger.WithError(err).Warn("Error during scrape")
 		return
 	}
 
 	for _, v := range volumeLists {
 		if v.Status.OwnerID == vc.currentNodeID {
+			var err error
+			var e *longhorn.Engine
+			var engineClientProxy engineapi.EngineClientProxy
+			var metrics *engineapi.Metrics
+
+			e, err = vc.ds.GetVolumeCurrentEngine(v.Name)
+			if err == nil {
+				engineClientProxy, err = vc.getEngineClientProxy(e)
+				if err == nil {
+					defer engineClientProxy.Close()
+
+					metrics, err = engineClientProxy.MetricsGet(e)
+					if err != nil {
+						vc.logger.WithError(err).Warnf("Failed to get metrics from volume %v from engine %v", e.Spec.VolumeName, e.Name)
+					}
+				} else {
+					vc.logger.WithError(err).Warnf("Failed to get engine for volume %v", v.Name)
```

Comment:
`Failed to get engine proxy of %v for volume %v ...`

Reply:
Updated. Thank you.